src/modules/Auth/Components/ChangePassword/ChangePassword.tsx

```diff
@@ -1,9 +1,318 @@
+      if (response.data.role === "admin") {
+        toast.success("Password changed successfully!");
+        navigate("/dashboard");
+      } else {
+        navigate("/booking"); // Navigate to home page for users
+      }
+    } catch (error: any) {
+      toast.error(error.response?.data.message || "Change Password failed");
+    }
+  };
+
+  //Handel show password
+  const [showOldPassword, setShowOldPassword] = React.useState(false);
+  const [showNewPassword, setShowNewPassword] = React.useState(false);
+  const [showConfirmNewPassword, setShowConfirmPassword] =
+    React.useState(false);
+
+  const handleClickShowOldPassword = () => setShowOldPassword((show) => !show);
+
+  const handleClickShowNewPassword = () => setShowNewPassword((show) => !show);
+
+  const handleClickShowConfirmNewPassword = () =>
+    setShowConfirmPassword((show) => !show);
+
+  const handleMouseDownPassword = (
+    event: React.MouseEvent<HTMLButtonElement>
+  ) => {
+    event.preventDefault();
+  };
+
+  const handleMouseUpPassword = (
+    event: React.MouseEvent<HTMLButtonElement>
+  ) => {
+    event.preventDefault();
+  };
+
+  const Item = styled(Paper)(({ theme }) => ({
+    ...theme.typography.body2,
+    textAlign: "left",
+    border: "none",
+    position: "relative",
+    overflow: "hidden",
+  }));
+
+  const OverlayText = styled(Typography)(({ theme }) => ({
+    position: "absolute",
+    bottom: "20%",
+    transform: "translateX(-50%)",
+    color: "white",
+    textAlign: "left",
+    padding: "10px",
+    borderRadius: "5px",
+  }));
+
   return (
-    <div>
-      ChangePassword
-    </div>
-  )
+    <Grid container sx={{ textAlign: "left" }}>
+      <Grid item xs={12} md={6}>
+        <Item sx={{ margin: "5%", boxShadow: "none" }}>
+          <img src={LogoImage} alt="ChangePassword" />
+        </Item>
+        <Item
+          sx={{
+            padding: {
+              xs: " 1.25rem",
+              sm: "3.125rem",
+              md: "6.25rem",
+              lg: "0.125rem 9.25rem",
+            },
+            boxShadow: "none",
+          }}
+        >
+          <Typography
+            sx={{
+              fontSize: "30px",
+              fontWeight: "bolder",
+              marginBottom: "10px",
+            }}
+          >
+            Change Password
+          </Typography>
+          <Typography></Typography>
+
+          <Typography sx={{ marginBottom: "2rem" }}>
+            If you already have an account registered{" "}
+            <Link
+              onClick={() => navigate("/login")}
+              underline="none"
+              sx={{
+                color: "var(--primary-color)",
+                fontWeight: "bolder",
+                cursor: "pointer",
+              }}
+            >
+              {" "}
+              Login here!
+            </Link>
+          </Typography>
+
+          <form onSubmit={handleSubmit(onSubmit)}>
+            {/* Old Password */}
+            <Typography
```

Comment:
We can create a password component with all props we need and include all the logic related to it in the component. This will make our code not repeated and will give us a simpler component api with the things that we want to change.